misc/testerina/modules/testerina-compiler-plugin/src/main/java/org/ballerinalang/testerina/compiler/TestExecutionModificationTask.java

```diff
@@ -0,0 +1,88 @@
+/*
+ *  Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.testerina.compiler;
+
+import io.ballerina.compiler.syntax.tree.FunctionDefinitionNode;
+import io.ballerina.compiler.syntax.tree.ImportDeclarationNode;
+import io.ballerina.compiler.syntax.tree.MethodCallExpressionNode;
+import io.ballerina.compiler.syntax.tree.ModuleMemberDeclarationNode;
+import io.ballerina.compiler.syntax.tree.ModulePartNode;
+import io.ballerina.compiler.syntax.tree.NodeList;
+import io.ballerina.compiler.syntax.tree.StatementNode;
+import io.ballerina.compiler.syntax.tree.SyntaxTree;
+import io.ballerina.projects.Document;
+import io.ballerina.projects.DocumentId;
+import io.ballerina.projects.Module;
+import io.ballerina.projects.ModuleId;
+import io.ballerina.projects.plugins.ModifierTask;
+import io.ballerina.projects.plugins.SourceModifierContext;
+import io.ballerina.tools.text.TextDocument;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Code modification task to generate the main Testerina runtime function.
+ *
+ * @since 2201.3.0
+ */
+public class TestExecutionModificationTask implements ModifierTask<SourceModifierContext> {
+
+    @Override
+    public void modify(SourceModifierContext modifierContext) {
+
+        if (!TesterinaCompilerPluginUtils.isSingleFileProject(modifierContext.currentPackage().project())) {
```

Comment:
We don't need a separate util for this IMO. We can use `modifierContext.currentPackage().project().equals(ProjectKind.SINGLE_FILE_PROJECT)`